src/bam/mod.rs

```diff
@@ -358,6 +358,20 @@ impl Writer {
         Ok(Writer { f: f, header: HeaderView::new(header_record) })
     }
 
+    /// Activate multi-threaded BAM write support in htslib. This should permit faster
+    /// writing of large BAM files.
+    /// # Arguments
+    ///
+    /// * `n_threads` - number of background writer threads to use
+    pub fn set_threads(&mut self, n_threads: usize) -> Result<(), WriteError> {
```

Comment:
Maybe have a separate error type for this? ThreadingError::Some? With a description like "error setting threads for multithreaded writing".